mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp

```diff
@@ -19,14 +19,22 @@ using mlir::arith::ConstantOp;
 
 namespace mlir {
 namespace miopen {
-//===----------------------------------------------------------------------===//
-// Utility function to emit constant float op. Returns a scalar.
-//===----------------------------------------------------------------------===//
+namespace {
+Value createConstantI8Op(OpBuilder &b, Location loc, Type type,
+                         Type elementType, int8_t value) {
+  Value retValue;
+  if (auto vecType = type.dyn_cast<VectorType>()) {
+    retValue =
+        b.create<ConstantOp>(loc, SplatElementsAttr::get(vecType, value));
+  } else {
+    retValue = b.create<ConstantOp>(loc, b.getI8IntegerAttr(value), type);
+  }
+
+  return retValue;
+}
+
```

Comment:
Why not expose this? We might want it in miopen-gen.cpp later

Reply:
Just trying to follow the defensive programming style that we can expose it when we need it. It's going to be a fairly trivial change. By then we can judge how and what to expose as public.

But right now since there's no consumer I'm just going to hide the implementation in this compilation unit only.

Reply:
Makes total sense